aexpect/remote_door.py

```diff
@@ -0,0 +1,718 @@
+"""
+
+SUMMARY
+------------------------------------------------------
+Share code between remote locations.
+
+Copyright: Intra2net AG
+
+
+CONTENTS
+------------------------------------------------------
+
+Utility for sharing code among multiple remote platforms, executing parts of a
+program on different locations (distributed programming) or writing controls
+executed on another host (metaprogramming).
+
+We use the following three options to communicate code among remote hosts:
+
+1) Static (template) controls are the classical way we use to do this,
+   namely by metaprogramming and setting attributes on a remote control
+   file and then running the control on the remote location.
+
+2) Generated controls / remote utilities are fully programmatically made
+   but simple (a few lines long) ways to perform a single call to a module
+   or utility installed on another machine (virtual or remote physical).
+
+3) Remote objects are python code objects provided over the network that
+   allow data persistence throughout multiple calls but are limited to
+   serialization support. Ultimately, one could share greater functionality
+   persistently over the network by extending the usual classes with ones
+   that are more serialization compatible (using pyro proxy instances).
+
+
+INTERFACE
+------------------------------------------------------
+
+"""
+
+import os
+import re
+import logging
+import inspect
+import threading
+import tempfile
+import time
+# NOTE: enable this before importing the Pyro backend in order to debug issues
+# related to connectivity and perform further development on the this utility.
+# os.environ["PYRO_LOGLEVEL"] = "DEBUG"
+import Pyro4
+
+try:
+    from aexpect import remote
+except ImportError:
+    pass
+
+
+#############################################
+### REMOTE UTILITIES / GENERATED CONTROLS ###
+#############################################
+
+#: default location for template control files
+SRC_CONTROL_DIR = "."
+#: default location for dumped control files
+DUMP_CONTROL_DIR = "."
+#: python binary to use for remote door running
+REMOTE_PYTHON_BINARY = "/usr/bin/python3.7"
```

Comment:
The reason was rather historical as we wanted to refer to a specific python binary on systems where there are multiple python versions and but not so much different installs. How about changing this value to `python3`? We could then set it to needed values from outside by setting the module attribute here, the default will then remain as simple and general as possible.

Reply:
`python3` as the default is reasonable. Does it make sense to make it configurable, ie, through env variables?

Reply:
I think adding env variables into the game might overcomplicate things, mostly because we are talking about remote execution and part of the configuration is local and part remote. The classical way of setting a module attribute makes more sense in this case. It is still configurable and clearly identified by the `REMOTE_` prefix.